cmake/FindHiopHipLibraries.cmake

```diff
@@ -0,0 +1,52 @@
+#[[
+
+Exports target `hiop_hip` which finds all hip libraries needed by hiop.
+
+#]]
+
+add_library(hiop_hip INTERFACE)
+
+find_library(HIPBLAS_LIBRARY
+  NAMES
+  hipblas
+  PATHS
+  ENV LD_LIBRARY_PATH ENV DYLD_LIBRARY_PATH
+  PATH_SUFFIXES
+  lib64 lib)
+
+if(HIPBLAS_LIBRARY)
+  get_filename_component(HIPBLAS_LIBRARY_DIR ${HIPBLAS_LIBRARY} DIRECTORY)
+endif()
+
+find_path(HIPBLAS_INCLUDE_DIR
+  NAMES
+  hipblas.h
+  PATHS
+  ${HIPBLAS_LIBRARY_DIR}/..
+  PATH_SUFFIXES
+  include)
+
+if(HIPBLAS_LIBRARY)
+  add_library(Hipblas INTERFACE)
+  target_link_libraries(Hipblas INTERFACE ${HIPBLAS_LIBRARY})
+  target_include_directories(Hipblas INTERFACE ${HIPBLAS_INCLUDE_DIR})
+  message(STATUS "Found Hipblas include: ${HIPBLAS_INCLUDE_DIR}")
+  message(STATUS "Found Hipblas library: ${HIPBLAS_LIBRARY}")
+else()
+  message(STATUS "Hipblas was not found.")
+endif()
+
+# for now we rely on MAGMA for GPUs computations
+include(FindHiopMagma)
+target_link_libraries(Magma    INTERFACE Hipblas)
+
+# Find hip cmake targets
+find_package(hip REQUIRED)
```

Comment:
This is where we should add something in `cmake/HiOpConfig.cmake.in`. Thankfully it is simple in this case. A consuming project will error when linking against HiOp here with `hip::device Magma` not found. Now, if the project has `find_package(hip REQUIRED)` somewhere in their CMake then this is no issue.

In [cmake/HiopConfig.cmake.in](https://github.com/LLNL/hiop/blob/develop/cmake/HiOpConfig.cmake.in#L28), currently there is logic for exporting the CUDA config that depends on `HIOP_USE_GPU`, and this should be modified to be similar to main CMakeLists.txt in terms of logic + any additional configurations:

```CMake
if(@HIOP_USE_GPU@) 
  if(@HIOP_USE_CUDA@)
    # Existing CUDA config
  endif()
  if(@HIOP_USE_HIP@)
    find_package(hip REQUIRED)
    add_definitions(-DHAVE_HIP) # For hipmagma
  endif()
endif()
```